aspnetcore/security/authentication/configure-oidc-web-authentication.md

```diff
@@ -0,0 +1,317 @@
+
+Implement a default sign-out page and change the Logout razor page code with this:
+
+```csharp
+[Authorize]
+public class LogoutModel : PageModel
+{
+    public IActionResult OnGetAsync()
+    {
+        return SignOut(new AuthenticationProperties
+        {
+            RedirectUri = "/SignedOut"
+        },
+        CookieAuthenticationDefaults.AuthenticationScheme,
+        OpenIdConnectDefaults.AuthenticationScheme);
+    }
+}
+```
+
+The `SignedOut.cshtml` requires the AllowAnonymous attribute.
+
+```csharp
+[AllowAnonymous]
+public class SignedOutModel : PageModel
+{
+    public void OnGet()
+    {
+    }
+}
+```
+
+### Add a login, logout button for the user.
+
+```
+@if (Context.User.Identity!.IsAuthenticated)
+{
+	<li class="nav-item">
+		<a class="nav-link text-dark" asp-area="" asp-page="/Logout">Logout</a>
+	</li>
+
+	<span class="nav-link text-dark">Hi @Context.User.Identity.Name</span>
+}
+else
+{
+	<li class="nav-item">
+		<a class="nav-link text-dark" asp-area="" asp-page="/Index">Login</a>
+	</li>
+}
+```
+
+## Examples with code snippets
+
+### Example Using User Info endpoint
+
+The OpenID Connect options can be used to map claims, implement handlers or even save the tokens in the session for later usage. 
+
+The **Scope** option can be used to request different claims or a refresh token which is sent as information to the OpenID Connect server. Requesting the **offline_access** is asking the server to return a reference token which can be used to refresh the session without authenticating the user of the application again.
+
+```csharp
+services.AddAuthentication(options =>
+{
+    options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+    options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+})
+.AddCookie()
+.AddOpenIdConnect(OpenIdConnectDefaults.AuthenticationScheme, options =>
+{
+    var oidcConfig = builder.Configuration.GetSection("OpenIDConnectSettings");
+    options.Authority = oidcConfig["IdentityProviderUrl"];
+    options.ClientSecret = oidcConfig["ClientSecret"];
+    options.ClientId = oidcConfig["Audience"];
+    options.ResponseType = OpenIdConnectResponseType.Code;
+
+    options.Scope.Clear();
+    options.Scope.Add("openid");
+    options.Scope.Add("profile");
+    options.Scope.Add("email");
+    options.Scope.Add("offline_access");
+
+    options.ClaimActions.Remove("amr");
+    options.ClaimActions.MapJsonKey("website", "website");
+
+    options.GetClaimsFromUserInfoEndpoint = true;
+    options.SaveTokens = true;
+
+    // .NET 9 feature
+    options.PushedAuthorizationBehavior = PushedAuthorizationBehavior.Require;
+
+    options.TokenValidationParameters.NameClaimType = JwtRegisteredClaimNames.Name;
+    options.TokenValidationParameters.RoleClaimType = "roles";
```

Comment:
Isn't "role" more common? I also don't think the constant adds much value.

```suggestion
    options.TokenValidationParameters.NameClaimType = "name";
    options.TokenValidationParameters.RoleClaimType = "role";
```

Reply:
fixed